cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -56,7 +56,8 @@ func ValidateMasterConfiguration(c *kubeadm.MasterConfiguration) field.ErrorList
 	allErrs = append(allErrs, ValidateCertSANs(c.Etcd.ServerCertSANs, field.NewPath("etcd").Child("serverCertSANs"))...)
 	allErrs = append(allErrs, ValidateCertSANs(c.Etcd.PeerCertSANs, field.NewPath("etcd").Child("peerCertSANs"))...)
 	allErrs = append(allErrs, ValidateAbsolutePath(c.CertificatesDir, field.NewPath("certificatesDir"))...)
-	allErrs = append(allErrs, ValidateNodeName(c.NodeName, field.NewPath("nodeName"))...)
+	allErrs = append(allErrs, ValidateNodeName(c.NodeRegistration.Name, field.NewPath("nodeRegistration").Child("name"))...)
```

Comment:
What about creating `validateNodeRegistrationOptions` ?

Reply:
This is WIP, thanks for keeping me honest fixing this as well